test/io/request_test.dart

```diff
@@ -10,61 +10,48 @@ import 'package:test/test.dart';
 import 'utils.dart';
 
 void main() {
-  test('.send', () {
-    expect(
-        startServer().then((_) {
-          var request = http.Request('POST', serverUrl)
-            ..body = 'hello'
-            ..headers['User-Agent'] = 'Dart';
+  setUp(startServer);
 
-          expect(
-              request.send().then((response) {
-                expect(response.statusCode, equals(200));
-                return response.stream.bytesToString();
-              }).whenComplete(stopServer),
-              completion(parse(equals({
-                'method': 'POST',
-                'path': '/',
-                'headers': {
-                  'content-type': ['text/plain; charset=utf-8'],
-                  'accept-encoding': ['gzip'],
-                  'user-agent': ['Dart'],
-                  'content-length': ['5']
-                },
-                'body': 'hello'
-              }))));
-        }),
-        completes);
-  });
+  tearDown(stopServer);
+
+  test('.send', () async {
+    final request = http.Request('GET', serverUrl)
+      ..body = 'hello'
+      ..headers['User-Agent'] = 'Dart';
+
+    final response = await request.send();
 
-  test('#followRedirects', () {
+    expect(response.statusCode, equals(200));
+    final bytesString = await response.stream.bytesToString();
     expect(
-        startServer().then((_) {
-          var request = http.Request('POST', serverUrl.resolve('/redirect'))
-            ..followRedirects = false;
-          var future = request.send().then((response) {
-            expect(response.statusCode, equals(302));
-          });
-          expect(
-              future.catchError((_) {}).then((_) => stopServer()), completes);
-          expect(future, completes);
-        }),
-        completes);
+        bytesString,
+        parse(equals({
+          'method': 'GET',
+          'path': '/',
+          'headers': {
+            'content-type': ['text/plain; charset=utf-8'],
+            'accept-encoding': ['gzip'],
+            'user-agent': ['Dart'],
+            'content-length': ['5']
+          },
+          'body': 'hello',
+        })));
+  });
+
+  test('#followRedirects', () async {
```

Comment:
nit: change the name while you are at it? `#followRedirects = false` or something? It is testing the opposite of what it sounds like right now.

Reply:
Maybe there should be a real follow redirects test also?

Reply:
sure, I can add one in a followup